pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -118,6 +118,10 @@ func (r *pullthroughBlobStore) proxyStat(ctx context.Context, retriever importer
 }
 
 // ServeBlob attempts to serve the requested digest onto w, using a remote proxy store if necessary.
+// Important! This function is called for GET and HEAD requests. Docker client uses[1] HEAD request
```

Comment:
@legionus ok I think that makes sense for the mirroring case (I want to mirror the full image with all blobs when I push)